src/uwtools/rocoto.py

```diff
@@ -0,0 +1,44 @@
+"""
+Support for writing a Rocoto XML using J2Templating
+"""
+
+from uwtools.j2template import J2Template
+from uwtools.config import YAMLConfig
+
+# private functions
+def _add_jobname(tasks: dict) -> None:
+    """
+    Add the jobname entry for all the tasks in the workflow.
+
+    :param tasks: Dict of tasks in workflow.
+    """
+    if not isinstance(tasks, dict):
+        return
+    for task, task_settings in tasks.items():
+        task_type = task.split("_", maxsplit=1)[0]
+        if task_type == "task":
+            # Use the provided attribute if it is present, otherwise use
+            # the name in the key
+            tasks[task]["jobname"] = \
+                task_settings.get("attrs", {}).get("name") or \
+                task.split("_", maxsplit=1)[1]
```

Comment:
The expression `task.split("_", maxsplit=1)` is repeated a few lines up. Could we assign this value to a variable and then reference that variable in both places?

I realize that this code was copied from SRW, but now's our best chance to improve it a bit.